File: find_tankup_rhp.py
import requests
from bs4 import BeautifulSoup
import PyPDF2
import io
import re

url = "https://www.hemsecurities.com/investor-relations/offer-documents"
try:
    response = requests.get(url, headers={'User-Agent': 'Mozilla/5.0'})
    soup = BeautifulSoup(response.text, 'html.parser')
    links = soup.find_all('a', href=True)
    tankup_link = None
    for link in links:
        if 'tankup' in link.text.lower() or 'tankup' in link['href'].lower():
            if 'rhp' in link.text.lower() or 'red' in link.text.lower() or 'rhp' in link['href'].lower():
                tankup_link = link['href']
                break
    
    if tankup_link:
        if not tankup_link.startswith('http'):
            tankup_link = 'https://www.hemsecurities.com' + tankup_link
        print(f"FOUND TANKUP RHP: {tankup_link}")
    else:
        print("Tankup RHP not found on HEM Securities.")
except Exception as e:
    print(f"Error checking HEM Securities: {e}")

File: find-tankup-rhp.js
const cheerio = require('cheerio');
const { readDB, writeDB } = require('./db.js');
const execSync = require('child_process').execSync;

async function run() {
    const url = "https://www.hemsecurities.com/investor-relations/offer-documents";
    try {
        const response = await fetch(url, { headers: { 'User-Agent': 'Mozilla/5.0' } });
        const html = await response.text();
        const $ = cheerio.load(html);
        let tankupLink = null;
        
        $('a').each((i, el) => {
            const href = $(el).attr('href') || '';
            const text = $(el).text().toLowerCase();
            if (href.includes('tankup') || text.includes('tankup')) {
                if (href.includes('rhp') || text.includes('rhp') || text.includes('red')) {
                    tankupLink = href;
                }
                // fallback to any pdf
                if (!tankupLink && href.endsWith('.pdf')) tankupLink = href;
            }
        });
        
        if (tankupLink) {
            if (!tankupLink.startsWith('http')) {
                tankupLink = 'https://www.hemsecurities.com' + (tankupLink.startsWith('/') ? '' : '/') + tankupLink;
            }
            console.log(`FOUND TANKUP RHP: ${tankupLink}`);
            
            const db = readDB();
            const tankup = db.companies.find(c => c.companyName.includes('Tankup'));
            if (tankup) {
                tankup.rhpUrl = tankupLink;
                console.log("Extracting Pre-IPO from RHP...");
                const pyCmd = `venv/bin/python nlp_extractor.py --rhp "${tankupLink}"`;
                const out = execSync(pyCmd, { encoding: 'utf8', stdio: ['pipe', 'pipe', 'ignore'], timeout: 60000 });
                const nlpData = JSON.parse(out.trim());
                tankup.preIpoInvestors = nlpData.preIpoInvestors || [];
                console.log("Investors:", tankup.preIpoInvestors);
                writeDB(db);
                console.log("DB updated!");
            }
        } else {
            console.log("Tankup RHP not found on HEM Securities.");
        }
    } catch (e) {
        console.error("Error:", e.message);
    }
}
run();
